### DISCUSSION TITLE
GPU not updating quick and FPS not showing on AMD GPU

### DISCUSSION THREAD

**Discussion by DylanGegax** [mathoudebine/turing-smart-screen-python#536] [Q&A]
Hi! I really love this theme [here](https://github.com/mathoudebine/turing-smart-screen-python/discussions/367) just I'm running into some problems on the GPU side of things. The CPU updates quick like it should but the GPU stats for some reason take like 4 mins to update. The default theme on the program works fine with updating GPU stats close to real time so I'm not sure what's wrong. Also the FPS stat isn't working either even though I have a AMD GPU. I'd really appreciate any help I could get to fixing this :)

**Comment by DylanGegax** [comment]
> Hi! I really love this theme [here](https://github.com/mathoudebine/turing-smart-screen-python/discussions/367) just I'm running into some problems on the GPU side of things. The CPU updates quick like it should but the GPU stats for some reason take like 4 mins to update. The default theme on the program works fine with updating GPU stats close to real time so I'm not sure what's wrong. Also the FPS stat isn't working either even though I have a AMD GPU. I'd really appreciate any help I could get to fixing this :)

Okay so I fixed the GPU not updating quick by going into the themes YAML file and changing the interval stuff to where instead of each part having its own interval, there's only one set interval for all of the GPU stats. Still need help with the FPS not working though :(

**Reply by c0ldJS** [reply]
Currently (based upon the table on [this page](https://github.com/mathoudebine/turing-smart-screen-python/wiki/System-monitor-%3A-hardware-support)), FPS monitoring is only available to Windows flavored machines that use `LibreHardwareMonitor` to shuttle metrics. Does that combination fit the setup you've got?

**Reply by DylanGegax** [reply]
I believe so, I do have a AMD RX 6600 and Windows. As for Libre I'm not sure but it does come up as an option just when I select it, it still doesn't work sadly :(

![FPS Libre Monitor](https://github.com/user-attachments/assets/f413a485-6de8-400e-80c1-9f5a1a0d6137)

**Reply by c0ldJS** [reply]
Inside your copy of `turing-smart-screen-python` is a `~\external\LibreHardwareMonitor\` directory and, in there, the `test_librehardwaremonitor.py` tool. Run that on the command line (as administrator) and it'll spit out everything that LHM can determine about your machine.

You'll want to look and confirm that there's a `GpuAmd` section with a `FPS` listing (and, of course, some sort of framerate value).

Does your CPU have an integrated GPU? If so, are you certain that `turing-smart-screen-python` is returning the AMD particulars and not the iGPU?

**Reply by c0ldJS** [reply]
Also, if you're on version 3.5.0 of this project then you might want to have a look at issue #530 as it might be also talkin' about what you're running into.

**Reply by DylanGegax** [reply]
This is what I got after running that tool and this is what popped up so I believe that parts good? Also nada I have a Intel i5 10400f so no iGPU. I do have version 3.5.0 but a different GPU. Any suggestions for further trouble shooting?? Thank you btw for all the help!!

Edit: My motherboard is also different as its a MSI Z490-A Pro

![gpuAMD command](https://github.com/user-attachments/assets/712158ca-4930-464c-aaf5-75e99a6fcf49)

**Reply by c0ldJS** [reply]
#530 is less about hardware combinations and a little bit more to do with something happening with graphics adapter detection between 3.4.0 and 3.5.0 .

When your display is using the theme, does the area for FPS show nothing, "`0`", or "`-1`"? If it's reading "`-1`" then it's matching what LHM is getting back from its polling (have a look at the `Factor | Fullscreen FPS` line).

**Reply by DylanGegax** [reply]
> #530 is less about hardware combinations and a little bit more to do with something happening with graphics adapter detection between 3.4.0 and 3.5.0 .
> 
> 
> 
> When your display is using the theme, does the area for FPS show nothing, "`0`", or "`-1`"? If it's reading "`-1`" then it's matching what LHM is getting back from its polling (have a look at the `Factor | Fullscreen FPS` line).

It's showing "0". Anything I should do when it just shows that?

**Reply by mathoudebine** [reply]
The GPU "Fullscreen FPS" sensor from LibreHardwareMonitor returns -1 when the GPU is not in use. The FPS sensor is only available during games / video enconding / etc. This is a limitation from the AMD drivers.

When the sensor returns "-1" the Turing Smart Screen program treats it as "No new value available" and displays the latest FPS value instead. If the program just started then 0 will be displayed.

Can you try running a video game or maybe a GPU benchmark ? The FPS value should start to update, and will freeze when the game/benchmark stops

**Reply by DylanGegax** [reply]
Hi! Sadly, it just shows "0" all the time even with a game running :(

**Reply by c0ldJS** [reply]
> The GPU "Fullscreen FPS" sensor from LibreHardwareMonitor returns -1 when the GPU is not in use. The FPS sensor is only available during games / video enconding / etc.

This is super-helpful to know. Thanks for exposing this behavior.

> Hi! Sadly, it just shows "0" all the time even with a game running

Don't forget that you can run that `test_librehardwaremonitor.py` tool while throwing work at the GPU (or whatever test upon localhost in general) to get a visual on the polling metrics or gauge whether the situation is meeting an expectation.

Is it safe to assume that the GPU interval you changed in the theme is faster, than, say, 10sec?
